message/message.yml

```diff
@@ -0,0 +1,412 @@
+    get:
+      tags:
+        - Message Queue
+      summary: Get the current queue size
+      description: Gets the current queue size (in bytes). This is helpful for determining the amount of memory that unread messages are consuming.
+      operationId: GetCurrentQueueSIze
+      responses:
+        200:
+          schema:
+            $ref: '#/definitions/QueueSizeBytes'
+          description: Success
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/sessions/{token}/configuration/maxQueueSizeBytes:
+    parameters:
+      - $ref: '#/parameters/Token'
+    get:
+      tags:
+        - Message Queue
+      summary: Get the maximum queue size
+      description: Gets the maximum queue size (in bytes). This is helpful for determining how much total memory is available for storing messages.
+      operationId: GetMaximumQueueSize
+      responses:
+        200:
+          schema:
+            $ref: '#/definitions/MaxQueueSizeBytes'
+          description: Success
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/websocket:
+    get:
+      tags:
+        - WebSocket (SystemLink Server only)
+      summary: Open a WebSocket session
+      description: Opens a persistent connection to the web server that allows bi-directional communication using a JSON
+                   formatted protocol. Once the connection is open, available actions include the WebSocket Publish Action,
+                   WebSocket Subscribe Action, and WebSocket Unsubscribe Action. The schema for these actions is defined in
+                   the *Models* section. Note that WebSocket functionality is currently only supported in SystemLink Server.
+      operationId: OpenWebSocketSession
+      responses:
+        101:
+          description: Switching Protocols
+          headers:
+            Sec-WebSocket-Accept:
+              description: Used in the WebSocket opening handshake. Set to a value derived from the Sec-WebSocket-Key header.
+              type: string
+        426:
+          description: Upgrade Required
+        default:
+          $ref: '#/responses/Error'
+      parameters:
+        - in: header
+          name: Upgrade
+          type: string
+          default: websocket
+        - in: header
+          name: Connection
+          type: string
+          default: Upgrade
+        - in: header
+          name: Sec-WebSocket-Version
+          description: A value used in the WebSocket opening handshake. Indicates the protocol version of the connection.
+          type: string
+        - in: header
+          name: Sec-WebSocket-Key
+          description: A value used in the WebSocket opening handshake. Provides part of the information used by the server
+             to prove that it received a valid WebSocket opening handshake.
+          type: string
+          required: true
```

Comment:
All of these headers are required for a successful websocket upgrade. I would say remove the default or replace them with x-example.

If we want to be pedantic about it (and I'm not sure we need to), the 101 response must also contain the Upgrade and Connection headers.

Reply:
Replaced default with x-example.

I'll leave the 101 headers as is, since we're not sure if it's worth defining them.